#6375 breaks compilation on Cray

Since #6375 I get: 

```
[ 46%] Building CXX object source/base/CMakeFiles/obj_base_release.dir/mpi.cc.o
In file included from /users/ag/lib/dealii-master/include/deal.II/base/mpi.h:19:0,
                 from /users/ag/lib/dealii-master/source/base/mpi.cc:17:
/users/agrayver/lib/dealii-master/build/include/deal.II/base/config.h:330:30: error: operator '*' has no left operand
  ((DEAL_II_MPI_VERSION_MAJOR * 100 + \
                              ^
/users/ag/lib/dealii-master/source/base/mpi.cc:100:5: note: in expansion of macro 'DEAL_II_MPI_VERSION_GTE'
 #if DEAL_II_MPI_VERSION_GTE(3, 0)
     ^
```

After digging a bit, it turns that FindMPI.cmake does not export MPI_VERSION_MAJOR and MPI_VERSION_MINOR, so I have this in the config.h:

```
#ifdef DEAL_II_WITH_MPI
#  define DEAL_II_MPI_VERSION_MAJOR 
#  define DEAL_II_MPI_VERSION_MINOR 

#  define DEAL_II_MPI_VERSION_GTE(major,minor) \
 ((DEAL_II_MPI_VERSION_MAJOR * 100 + \
    DEAL_II_MPI_VERSION_MINOR) \
    >=  \
    (major)*100 + (minor))
#endif
```

This happens on a Cray cluster with their environment: 
https://pubs.cray.com/content/00477539-DA/DA00477538

--- a/cmake/modules/FindMPI.cmake
+++ b/cmake/modules/FindMPI.cmake
@@ -109,20 +109,19 @@ IF(NOT MPI_MPI_H MATCHES "-NOTFOUND" AND NOT DEFINED MPI_VERSION)
     MPI_VERSION_MINOR "${MPI_VERSION_MINOR_STRING}"
     )
   SET(MPI_VERSION "${MPI_VERSION_MAJOR}.${MPI_VERSION_MINOR}")
+ENDIF()
 
-  #
-  # Except - this doesn't always work. Some distributions install a header
-  # stub mpi.h that includes the right mpi header depending on the
-  # architecture. In this case we are really out of luck. It is not
-  # straightforward to find the correct header file to query the version
-  # information from. Just set a very conservative default:
-  #
-  IF(MPI_VERSION STREQUAL ".")
-    SET(MPI_VERSION "0.0")
-    SET(MPI_VERSION_MAJOR "0")
-    SET(MPI_VERSION_MINOR "0")
-  ENDIF()
-
+#
+# Except - this doesn't always work. Some distributions install a header
+# stub mpi.h that includes the right mpi header depending on the
+# architecture. In this case we are really out of luck. It is not
+# straightforward to find the correct header file to query the version
+# information from. Just set a very conservative default:
+#
+IF(NOT DEFINED MPI_VERSION OR MPI_VERSION STREQUAL ".")
+  SET(MPI_VERSION "0.0")
+  SET(MPI_VERSION_MAJOR "0")
+  SET(MPI_VERSION_MINOR "0")
 ENDIF()
 
 DEAL_II_PACKAGE_HANDLE(MPI

